homeassistant/components/overkiz/binary_sensor.py

```diff
@@ -40,64 +41,64 @@ class OverkizBinarySensorDescription(
         key=OverkizState.CORE_RAIN,
         name="Rain",
         icon="mdi:weather-rainy",
-        value_fn=lambda state: state == OverkizCommandParam.DETECTED,
+        value_fn=lambda state: state == cast(str, OverkizCommandParam.DETECTED),
```

Comment:
This isn't a string by an enum, hence the cast is needed here.

Instead, use:

```py
OverkizCommandParam.DETECTED.value
```

Reply:
I'm not sure what is not the understand?
The casting isn't valid, as it returns an Enum, not an str.

Reply:
We declare it also a string: https://github.com/iMicknl/python-overkiz-api/blob/main/pyoverkiz/enums/command.py#L73 It’s why I don’t get it.

Reply:
@tetienne I think what Frenck tries to say is that if you want to match (==) a string to an str enum, that you need to use the value, not compare the enum itself.

I didn't know that this was required, but it makes sense looking at how you access the value of an enum.

Reply:
In my original testing, this also failed to satisfy mypy.
`value_fn=lambda state: state == OverkizCommandParam.DETECTED.value,`

@bdraco suggested it might be due to missing `py.typed` file. Has that been looked into?

Reply:
We are currently looking into these issues, by adding correct typing to pyoverkiz. For now let's hold on all Overkiz PR's and fix it nicely.